### PR TITLE
[1548] FE MNO

### PR DESCRIPTION
**Pull request by asmega**
### Context

- https://trello.com/c/LlvyVpmg/1548-mno-rollout-to-fe-by-weds-17th

### Changes proposed in this pull request

- Adds MNO to FEs
- `School#hide_mno?` has now been re-purposed. Previously these schools could not see the MNO, they now can. Instead the flag is used to determine if only a subset of mobile networks are permitted
- On manual requests remove excluded networks for FE
- On bulk upload show errors if user attempt to upload FE excluded network. we should be rejected and not persisted 

### Screenshots

image

### Guidance to review

- Probably want to side by side compare current school journey vs FE journey
- For FE flag is `hide_mno` set to true. as there are some FE with `hide_mno` false as they have under 16s
- Perform manual request
- School journey should see all available networks
- FE journey should see all networks minus giffgaff and o2
- Perform bulk upload
- School journey should accept all available networks
- FE should not accept giffgaff and o2 - see screenshot above
- There are also various copy changes thru the journeys

### CODE REVIEW

File: app/views/shared/internet/_mno_guidance.html.erb
Comment (fofr) [inline app/views/shared/internet/_mno_guidance.html.erb:12]:
```suggestion
<p class="govuk-body">giffgaff and O2 will not accept requests for students over the age of 16.</p>
```

Comment (asmega) [reply app/views/shared/internet/_mno_guidance.html.erb:12]: good spot i will change to read from model instead

File: app/models/school.rb
Comment (fofr) [inline app/models/school.rb:78]: I wonder if we should have something like this?

```ruby
def hide_networks_not_supporting_fe?
  hide_mno?
end```

`hide_mno` doesn't do the heavy lifting in explaining

File: app/views/school/internet/mobile/bulk_requests/new.html.erb
Comment (fofr) [inline app/views/school/internet/mobile/bulk_requests/new.html.erb:39]: This block should only show to schools/colleges etc that we know have students over 16

File: app/models/school.rb
Comment (asmega) [reply app/models/school.rb:78]: yea probably a good call. i did consider renaming the entire column. Aliasing seems a good middle ground

File: app/views/school/internet/mobile/bulk_requests/new.html.erb
Comment (asmega) [reply app/views/school/internet/mobile/bulk_requests/new.html.erb:39]: this can't be done at the moment and we will have to show for all. we simply do not have the data for it. there is a backlog ticket titled `import `StatutoryLowAge` from GIAS`. if we do that first we can use that as the toggle for this

File: app/models/mobile_network.rb
Comment (tonyheadford) [inline app/models/mobile_network.rb:17]: This feels a bit icky, I think it would've been better as a flag/data item on the model?

**Review by tonyheadford — APPROVED** [review]
Looks good, I've made a comment about the `excluded_fe_networks` but don't want that to hold this up.

File: app/models/mobile_network.rb
Comment (asmega) [reply app/models/mobile_network.rb:17]: sure, this works so i'll merge this now and make the change once everything is sorted.

**Pull request merged**
